document_merge_service/api/management/commands/dms_encrypt_templates.py

```diff
@@ -0,0 +1,57 @@
+from django.conf import settings
+from django.core.exceptions import ImproperlyConfigured
+from django.core.files.storage import storages
+from django.core.management.base import BaseCommand
+from tqdm import tqdm
+
+from document_merge_service.api.models import Template
+
+
+class Command(BaseCommand):
+    help = "Swaps plain text template content to encrypted content"
+
+    def handle(self, *args, **options):
+        if not settings.DMS_ENABLE_AT_REST_ENCRYPTION:
+            return self.stdout.write(
+                self.style.WARNING(
+                    "Encryption is not enabled. Skipping encryption of templates."
+                )
+            )
+
+        failed_templates = []
+
+        # flip between default and encrypted storage to have the correct parameters in the requests
+        encrypted_storage = storages.create_storage(settings.STORAGES["default"])
+        unencrypted_storage_setting = settings.STORAGES["default"]
+        if (
+            "OPTIONS" not in unencrypted_storage_setting
+            or "object_parameters" not in unencrypted_storage_setting["OPTIONS"]
+        ):
+            raise ImproperlyConfigured(
+                "Encryption is enabled but no object_parameters found in the storage settings."
+            )
+        del unencrypted_storage_setting["OPTIONS"]["object_parameters"]
+        unencrypted_storage = storages.create_storage(unencrypted_storage_setting)
+
+        query = Template.objects.all()
```

Comment:
How do you know you're not encrypting an already encrypted file/template?

Reply:
There is no way, you just have to assume everything is plain text or encrypted.

If its already encrypted the fetching wont work, as we dont send the ssec key.